api/grpcserver/grpcserver_tls_test.go

```diff
@@ -1,32 +1,175 @@
 package grpcserver
 
 import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"net"
 	"os"
 	"path/filepath"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"go.uber.org/zap/zaptest"
 )
 
 const (
-	caCert     = "testdata/ca.crt"
-	serverCert = "testdata/server.crt"
-	serverKey  = "testdata/server.key"
-	clientCert = "testdata/client.crt"
-	clientKey  = "testdata/client.key"
+	caCertName     = "ca.crt"
+	caKeyName      = "ca.key"
+	serverCertName = "server.crt"
+	serverKeyName  = "server.key"
+	clientCertName = "client.crt"
+	clientKeyName  = "client.key"
 )
 
-func launchTLSServer(tb testing.TB, services ...ServiceAPI) (Config, func()) {
-	pwd, err := os.Getwd()
+func genKeys(tb testing.TB) string {
+	dir := tb.TempDir() // to store the generated files
+
+	caKey, err := rsa.GenerateKey(rand.Reader, 4096)
```

Comment:
Would it make sense to factor this part out?

Reply:
The problem is that the code block creating the keys and certificates is always slightly different, I'll see what I can do.